### PR TITLE
Improve AssemblyNameInfo Fuzzer

### PR DESCRIPTION
**Pull request by adamsitnik**
I am going to be honest, I just stole some good ideas from @buyaa-n from https://github.com/dotnet/runtime/pull/107206

### CODE REVIEW

**Comment by adamsitnik** [comment]
@MihuBot fuzz TypeName

**Comment by adamsitnik** [comment]
@MihuBot fuzz TypeName

**Comment by adamsitnik** [comment]
@MihuBot fuzz AssemblyNameInfo

File: src/libraries/Fuzzing/DotnetFuzzing/Fuzzers/AssemblyNameInfoFuzzer.cs [hunk 1]
```diff
@@ -26,19 +30,51 @@ public void FuzzTarget(ReadOnlySpan<byte> bytes)
 
         private static void Test(PooledBoundedMemory<char> inputPoisoned)
```
Comment (buyaa-n) [inline src/libraries/Fuzzing/DotnetFuzzing/Fuzzers/AssemblyNameInfoFuzzer.cs:31]: NIT: could change the parameter to span as only span of the poisoned input is used in the method
```suggestion
        private static void Test(Span<char> span)
```

**Review by buyaa-n — APPROVED** [review]
LGTM

File: src/libraries/Fuzzing/DotnetFuzzing/Fuzzers/AssemblyNameInfoFuzzer.cs (hunk 1 above)
Comment (adamsitnik) [reply src/libraries/Fuzzing/DotnetFuzzing/Fuzzers/AssemblyNameInfoFuzzer.cs:31]: > NIT: could change the parameter to span as only span of the poisoned input is used in the method

If that is not a problem I am going to merge it as is and include this change in next PR.

**Comment by adamsitnik** [comment]
/ba-g it's #107285, but for some reason it's not recognized by the build analysis

**Pull request merged**